src/components/productdetail/VariantSelector.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <div v-if="product"
+       class="row select-row">
+    <ul class="list-inline"
+        data-test="variant-selector-list">
+      <li v-for="(values, name) in attributes"
+          :key="name">
+        <p class="text-uppercase"
+           data-test="attribute-name">
+          {{name}}
+        </p>
+        <AttributeSelect :product="product"
+                         :values="values"
+                         :sku="sku"
+                         :name="name"
+                         @updateCombi="updateSku"
+                         :selected="selected"
+                         :variantCombinations="variantCombinations" />
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+import gql from 'graphql-tag';
+import flatMap from 'lodash.flatmap';
+import AttributeSelect from './AttributeSelect.vue';
+
+export default {
+  components: { AttributeSelect },
+
+  props: {
+    sku: {
+      type: String,
+      required: true,
+    },
+  },
+
+  data: () => ({
+    product: null,
+    selected: {},
+  }),
+
+  watch: {
+    variantCombinations(value) {
+      value.forEach((variant) => {
+        if (variant.sku === this.sku) {
+          Object.assign(this.selected, variant);
+        }
+      });
+    },
+  },
+
+  methods: {
+    updateSku(selectedValues) {
+      Object.assign(this.selected, selectedValues);
+      this.$router.push({ path: selectedValues.sku });
+    },
+
+    groupValuesByAttribute(acc, currentItem) {
+      if (!acc[currentItem.name]) {
+        acc[currentItem.name] = [];
+      }
+      acc[currentItem.name].push(currentItem.value || currentItem.label);
+      return acc;
+    },
+  },
+
+  computed: {
+    attributes() {
+      const { allVariants } = this.product.masterData.current;
+      return flatMap(allVariants, variant => Object.values(variant.attributes)
+        .filter(attr => typeof attr === 'object'))
+        .reduce(this.groupValuesByAttribute, {});
+    },
+
+    variantCombinations() {
+      if (this.product) {
+        return this.product.masterData.current.allVariants
+          .map((variant) => {
+            const combi = { sku: variant.sku };
+            Object.keys(variant.attributes).forEach((key) => {
+              if (key !== '__typename') {
+                combi[key] = variant.attributes[key].label || variant.attributes[key].value;
+              }
```

Comment:
I think it should be possible to do `delete variant.attributes.__typename` right before the `forEach` and then remove the `if (key !== '__typename')` part. I would give it a try.

Reply:
I see that what we gain on one side we lost it in the other 😅
The positive about this solution is that it is the solution we use in other places, so at least it's not a complete lost.